src/main/java/com/iexec/commons/poco/chain/IexecHubAbstractService.java

```diff
@@ -815,17 +813,11 @@ public Optional<ChainDeal> getChainDeal(String chainDealId) {
 
         byte[] chainDealIdBytes = BytesUtils.stringToBytes(chainDealId);
         try {
-            Tuple9<String, String, BigInteger, String, String, BigInteger, String, String, BigInteger> dealPt1 =
-                    iexecHub.viewDealABILegacy_pt1(chainDealIdBytes).send();
-            Tuple6<BigInteger, byte[], String, String, String, String> dealPt2 =
-                    iexecHub.viewDealABILegacy_pt2(chainDealIdBytes).send();
-            Tuple6<BigInteger, BigInteger, BigInteger, BigInteger, BigInteger, BigInteger> config =
-                    iexecHub.viewConfigABILegacy(chainDealIdBytes).send();
+            IexecHubContract.Deal deal = iexecHub.viewDeal(chainDealIdBytes).send();
 
-
-            String appAddress = dealPt1.component1();
-            String datasetAddress = dealPt1.component4();
-            BigInteger categoryId = config.component1();
+            String appAddress = deal.app.pointer;
+            String datasetAddress = deal.dataset.pointer;
```

Comment:
Can `deal.app` or `deal.dataset` be null?

Reply:
I don't think they can be `null`. `dataset` address could at most be `0x0` (no dataset)
https://github.com/iExecBlockchainComputing/PoCo-dev/blob/develop/contracts/modules/delegates/IexecPoco1Delegate.sol#L117